go/vt/vtgate/planbuilder/operators/apply_join.go

```diff
@@ -224,9 +216,12 @@ func (aj *ApplyJoin) getJoinColumnFor(ctx *plancontext.PlanningContext, orig *sq
 	case deps.IsSolvedBy(both):
 		col = breakExpressionInLHSandRHS(ctx, e, TableID(aj.LHS))
 	default:
-		panic(vterrors.VT13002(sqlparser.String(e)))
+		panic(vterrors.VT13001(fmt.Sprintf("expression depends on tables outside this join: %s", sqlparser.String(e))))
 	}
 
+	col.GroupBy = addToGroupBy
```

Comment:
This is another bug that was not related to the issue being found, but it caused some plans to be missing group by expressions